torch/csrc/jit/codegen/cuda/scheduler.cpp

```diff
@@ -369,15 +370,16 @@ c10::optional<ReductionParams> scheduleReduction(
   // Find Reduction TensorView
   // TODO: This is making an assumption there is only one reduction
   // in a kernel.  This will not be true in the long run.
-  TensorView* red_tv = nullptr;
-  for (auto& expr : fusion->exprs(/*from_outputs_only*/ true)) {
-    if (expr->type() == ExprType::ReductionOp) {
+  // TensorView* red_tv = nullptr;
+  // for (auto& expr : fusion->exprs(/*from_outputs_only*/ true)) {
+  /*  if (expr->type() == ExprType::ReductionOp) {
       red_tv = expr->output(0)->as<TensorView>();
       break;
     }
-  }
+  }*/
   TORCH_INTERNAL_ASSERT(
       red_tv != nullptr, "Reduction TensorView wasn't found.");
+  const auto red_expr = fusion->origin(red_tv);
```

Comment:
Probably should assert red_tv has reduction axes, and red_expr is a reduction op to validate inputs you were given.

Reply:
Assertion added above!